common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -0,0 +1,183 @@
+    public static final String QUERY_PROMPT_VALUE = "login";
+    public static final String LOGIN_HINT = "login_hint";
+
+    public static final String LIB_ID_PLATFORM = "x-client-SKU";
+    public static final String LIB_ID_VERSION = "x-client-Ver";
+    public static final String LIB_ID_CPU = "x-client-CPU";
+    public static final String LIB_ID_OS_VER = "x-client-OS";
+    public static final String LIB_ID_DM = "x-client-DM";
+
+    /**
+     * Required.
+     */
+    private URL mAuthority;
+    /**
+     * Required value.
+     * <p>
+     * Passed in from ADAL/MSAL after authority verification.
+     */
+    private String mAuthorizationEndpoint;
+    /**
+     * Can be used to pre-fill the username/email address field of the sign-in page for the user, if you know their username ahead of time.
+     */
+    private String mLoginHint;
+    /**
+     * Correlation ID.
+     */
+    private UUID mCorrelationId;
+    /**
+     * Used to secure authorization code grants via Proof Key for Code Exchange (PKCE) from a native client.
+     */
+    private PkceChallenge mPkceChallenge;
+    /**
+     * Extra query parameters.
+     */
+    private String mExtraQueryParam;
+    /**
+     * The version of the calling library.
+     */
+    private String mLibraryVersion;
+
+
+    public URL getAuthority() {
+        return mAuthority;
+    }
+
+    public void setAuthority(final URL authority) {
+        mAuthority = authority;
+    }
+
+    public String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    public void setAuthorizationEndpoint(final String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public String getLoginHint() {
+        return mLoginHint;
+    }
+
+    public void setLoginHint(final String loginHint) {
+        mLoginHint = loginHint;
+    }
+
+    public UUID getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public void setCorrelationId(final UUID correlationId) {
+        mCorrelationId = correlationId;
+    }
+
+    public PkceChallenge getPkceChallenge() {
+        return mPkceChallenge;
+    }
+
+    public void setPkceChallenge(final PkceChallenge pkceChallenge) {
+        mPkceChallenge = pkceChallenge;
+    }
+
+    public String getExtraQueryParam() {
+        return mExtraQueryParam;
+    }
+
+    public void setExtraQueryParam(final String extraQueryParam) {
+        mExtraQueryParam = extraQueryParam;
+    }
+
+    public String getLibraryVersion() {
+        return mLibraryVersion;
+    }
+
+    public void setLibraryVersion(final String libraryVersion) {
+        mLibraryVersion = libraryVersion;
+    }
+
+    /**
+     * Return the start URL to load in the web view.
+     *
+     * @return String of start URL.
+     * @throws UnsupportedEncodingException
+     * @throws ClientException
+     */
+    public abstract String getAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException;
```

Comment:
This has already been declared in the superclass -- please remove and relocate javadoc